### PR TITLE
fix: the link for viewing sushiswap's pair analytics in swap dialog i…

### LINKED ISSUE #3701: [Bug] The link for viewing sushiswap's pair analytics in swap dialog is missing prefix.(only for BSC or Matic network)
### How to reproduce?

1.Open swap dialog, change to BSC or Matic network and choose sushiswap;
2.Make some swap pairs, for example Matic->Titan;
3.Click pair analytics to check.

### What happened?
The link is missing some prefix so an error occurs. Here is the link:
https://analytics.sushi.com/pairs/0xa79983daf2a92c2c902cd74217efe3d8af9fba2a

### What should happen?
If you add prefix "-bsc" or "-polygon" (analytics-{chain_name})in the link, the corresponding pages are showed. 
https://analytics-bsc.sushi.com/pairs/0x70a164a8eccee2b7a1873af71efdfa82904f937d
https://analytics-polygon.sushi.com/pairs/0xa79983daf2a92c2c902cd74217efe3d8af9fba2a
### Mask Version:

- [ ] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [x] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [x] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [x] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by lelenei**
…s missing prefix

closes #3701

### CODE REVIEW

**Review by Himself65 — DISMISSED** [review]

File: packages/maskbook/src/plugins/Trader/pipes.ts [hunk 1]
```diff
@@ -86,20 +87,28 @@ export function resolveTradeProviderLink(tradeProvider: TradeProvider) {
     }
 }
 
-export function resolveTradePairLink(tradeProvider: TradeProvider, address: string) {
+export function resolveTradePairLink(chainId: ChainId, tradeProvider: TradeProvider, address: string) {
+    const chainDetailed = getChainDetailed(chainId)
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/pipes.ts:91]:
```suggestion
    const networkType = getNetworkTypeFromChainId(chainId)
```

File: packages/maskbook/src/plugins/Trader/SNSAdaptor/balancer/TradeRouteHop.tsx [hunk 2]
```diff
@@ -25,6 +26,8 @@ export interface TradeRouteHopProps {
 export function TradeRouteHop(props: TradeRouteHopProps) {
     const { hop } = props
     const classes = useStyles()
+    const chainId = useChainId()
+    const networkType = getNetworkTypeFromChainId(chainId)
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/SNSAdaptor/balancer/TradeRouteHop.tsx:30]:
```suggestion
    const networkType = useNetworkType()
```

File: packages/maskbook/src/plugins/Trader/pipes.ts [hunk 3]
```diff
@@ -86,20 +87,22 @@ export function resolveTradeProviderLink(tradeProvider: TradeProvider) {
...
     switch (tradeProvider) {
         case TradeProvider.UNISWAP:
             return `https://v2.info.uniswap.org/pair/${address}`
         case TradeProvider.ZRX:
             return ''
         case TradeProvider.SUSHISWAP:
-            return `https://analytics.sushiswap.fi/pairs/${address}`
+            if (networkType === NetworkType.Binance) return `https://analytics-bsc.sushi.com/pairs/${address}`
+            else if (networkType === NetworkType.Polygon) return `https://analytics-polygon.sushi.com/pairs/${address}`
+            else return `https://analytics.sushi.com/pairs/${address}`
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/pipes.ts:99]:
```suggestion
            if (networkType === NetworkType.Binance) return `https://analytics-bsc.sushi.com/pairs/${address}`
            if (networkType === NetworkType.Polygon) return `https://analytics-polygon.sushi.com/pairs/${address}`
            return `https://analytics.sushi.com/pairs/${address}`
```

File: packages/maskbook/src/plugins/Trader/pipes.ts [hunk 4]
```diff
@@ -86,20 +87,22 @@ export function resolveTradeProviderLink(tradeProvider: TradeProvider) {
...
     switch (tradeProvider) {
         case TradeProvider.UNISWAP:
             return `https://v2.info.uniswap.org/pair/${address}`
         case TradeProvider.ZRX:
             return ''
         case TradeProvider.SUSHISWAP:
-            return `https://analytics.sushiswap.fi/pairs/${address}`
+            if (networkType === NetworkType.Binance) return `https://analytics-bsc.sushi.com/pairs/${address}`
+            if (networkType === NetworkType.Polygon) return `https://analytics-polygon.sushi.com/pairs/${address}`
+            return `https://analytics.sushi.com/pairs/${address}`
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/pipes.ts:99]:
```suggestion
            switch (networkType) {
                case NetworkType.Ethereum: return `https://analytics.sushi.com/pairs/${address}`
                case NetworkType.Binance: return `https://analytics-bsc.sushi.com/pairs/${address}`
                case NetworkType.Polygon: return `https://analytics-polygon.sushi.com/pairs/${address}`
                default: 
                    safeUnreachable(networkType)
                    return ''
            }
```

Comment (septs) [reply packages/maskbook/src/plugins/Trader/pipes.ts:99]: why not lookup table?

```ts
const prefixURLs: Record<NetworkType, string> = {
    // ....
}
prefixURLs[networkType] + '/' + address
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**